xonsh/built_ins.py

```diff
@@ -289,16 +289,34 @@ def run_subproc(cmds, captured=True):
                 stdin, _ = stdin.read(), stdin.close()
             # Redirect the output streams temporarily. merge with possible
             # return values from function.
-            new_stdout, new_stderr = StringIO(), StringIO()
-            with redirect_stdout(new_stdout), redirect_stderr(new_stderr):
+            if stdout is PIPE:
+                new_stdout, new_stderr = StringIO(), StringIO()
+                rtn = None
+                with redirect_stdout(new_stdout), redirect_stderr(new_stderr):
+                    rtn = alias(cmd[1:], stdin=stdin)
+                proxy_stdout, proxy_stderr = new_stdout.getvalue(), new_stderr.getvalue()
+                if isinstance(rtn, tuple):
+                    if len(rtn) >= 1 and rtn[0]:
+                        proxy_stdout += rtn[0]
+                    if len(rtn) >= 2 and rtn[1]:
```

Comment:
This behaviour with tuple return types seems strange to me.  I think if it returns a tuple then it should be of length-2, nothing else really makes sense...